### PR TITLE
Add cache-middleware to improve performance of repeated http-requests

### PR DESCRIPTION
**Pull request by jkrzefski**

### CODE REVIEW

**Comment by JoshuaBehrens** [comment]
@jkrzefski can we remove this with your implementation? https://github.com/niemand-online/heptaconnect-portal-amiibo/blob/c22a043063e623ca264acac5c3587b0c8d0685b9/src/Support/AmiiboApiClient.php#L21

**Comment by jkrzefski** [comment]
@JoshuaBehrens Yes, good point. I removed it.

File: src/Support/HttpCacheMiddleware.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+final class HttpCacheMiddleware implements HttpClientMiddlewareInterface
+{
+    private array $cache = [];
+
+    public function process(RequestInterface $request, ClientInterface $handler): ResponseInterface
+    {
+        $cacheKey = (string) $request->getUri();
```
Comment (JoshuaBehrens) [inline src/Support/HttpCacheMiddleware.php:24]: This should only happen for "read only" requests like GET and HEAD

**Review by JoshuaBehrens — CHANGES_REQUESTED** [review]

File: src/Support/HttpCacheMiddleware.php (hunk 1 above)
Comment (jkrzefski) [reply src/Support/HttpCacheMiddleware.php:24]: Done

**Review by JoshuaBehrens — APPROVED** [review]

**Pull request merged**